griptape/drivers/prompt/huggingface_pipeline_prompt_driver.py

```diff
@@ -40,43 +48,49 @@ class HuggingFacePipelinePromptDriver(BasePromptDriver):
         )
     )
 
-    def try_run(self, prompt_stack: PromptStack) -> TextArtifact:
-        messages = [self._prompt_stack_input_to_message(input) for input in prompt_stack.inputs]
+    def try_run(self, prompt_stack: PromptStack) -> PromptStackMessage:
+        messages = self._prompt_stack_to_messages(prompt_stack)
 
         result = self.pipe(
-            messages,
-            max_new_tokens=self.max_tokens,
-            tokenizer=self.tokenizer.tokenizer,
-            stop_strings=self.tokenizer.stop_sequences,
-            temperature=self.temperature,
-            do_sample=True,
+            messages, max_new_tokens=self.max_tokens, temperature=self.temperature, do_sample=True, **self.params
         )
 
         if isinstance(result, list):
             if len(result) == 1:
                 generated_text = result[0]["generated_text"][-1]["content"]
 
-                return TextArtifact(value=generated_text)
+                input_tokens = len(self.__prompt_stack_to_tokens(prompt_stack))
+                output_tokens = len(self.tokenizer.tokenizer.encode(generated_text))
+
+                return PromptStackMessage(
+                    content=[TextPromptStackContent(TextArtifact(generated_text))],
+                    role=PromptStackMessage.ASSISTANT_ROLE,
+                    usage=PromptStackMessage.Usage(input_tokens=input_tokens, output_tokens=output_tokens),
+                )
             else:
                 raise Exception("completion with more than one choice is not supported yet")
         else:
             raise Exception("invalid output format")
 
-    def try_stream(self, prompt_stack: PromptStack) -> Iterator[TextArtifact]:
+    def try_stream(self, prompt_stack: PromptStack) -> Iterator[DeltaPromptStackMessage | BaseDeltaPromptStackContent]:
         raise NotImplementedError("streaming is not supported")
 
     def prompt_stack_to_string(self, prompt_stack: PromptStack) -> str:
         return self.tokenizer.tokenizer.decode(self.__prompt_stack_to_tokens(prompt_stack))
 
-    def _prompt_stack_input_to_message(self, prompt_input: PromptStack.Input) -> dict:
-        return {"role": prompt_input.role, "content": prompt_input.content}
+    def _prompt_stack_to_messages(self, prompt_stack: PromptStack) -> list[dict]:
+        messages = []
+        for i in prompt_stack.messages:
+            if len(i.content) == 1:
+                messages.append({"role": i.role, "content": TextPromptStackContent(i.to_text_artifact())})
```

Comment:
Can you replace `TextPromptStackContent(i.to_text_artifact())` with just `i` (or equivalently `i.content`)?

I understand that this works because of `BasePromptStackContent. __str__`, but is that equivalent to just `i`, since  `PromptStackMessage` also has `__str__` (which calls `self.to_text_artifact().to_text()`)?

Reply:
Cleaned up a bit, let me know what you think.